packages/ui-app/src/types.d.ts

```diff
@@ -25,3 +21,15 @@ export type BaseContext = {
     }
   }
 };
+
+export type FormErrors = {
+  inputError: {
```

Comment:
No nested structures like these in the state please. This just seems, well deep, cumbersome and a maintenance nightmare.